MSI magento-engcom/magento#65:

larsroettig opened the pull request.

Commit:
MSI magento-engcom/magento#65:
- Add test and functionality to support `OutOfStock` SourceItem

Commit:
MSI magento-engcom/magento#66:
- Add the test case
- Impelment Adjust StockItem aggregation functionality to support Disabled Sources

naydav commented on app/code/Magento/InventoryApi/Test/_files/source_items_out_stock.php:
```diff
@@ -0,0 +1,69 @@
+<?php
```
Need to merge develop brunch and adapt tests according to possibility to pre-define id

naydav replied:
Need to simplify fixture files
https://github.com/magento-engcom/magento2/pull/76/files#diff-c8b20cde2aec460219d830973d4b6321
like in
https://github.com/magento-engcom/magento2/blob/develop/app/code/Magento/InventoryApi/Test/_files/source_items.php

Now it is possible after merge
https://github.com/magento-engcom/magento2/pull/79
We try to describe more details in our article
https://github.com/magento-engcom/magento2/wiki/The-first-step-towards-pre-generated-IDs.-And-how-this-will-improve-your-Integration-tests

Commit:
Merge remote-tracking branch 'origin/develop' into msi-out-of-stock

Commit:
MSI magento-engcom/magento#65:
- Bugfix for unit tests

naydav commented on app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php:
```diff
@@ -37,9 +39,18 @@ public function __construct(ResourceConnection $resourceConnection)
 
+        // find all enabled sources
+        $select = $connection->select()->from($sourceTable, [Source::SOURCE_ID])->where(Source::ENABLED . '=?', 1);
+        if (count($sourceIds)) {
```
Need to check this condition
If we do not have enabled sources then we need to return empty data

naydav commented on app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php:
```diff
@@ -37,9 +39,18 @@ public function __construct(ResourceConnection $resourceConnection)
+        if (count($sourceIds)) {
+            $select->where(Source::SOURCE_ID . ' = ?', $sourceIds);
+        }
+        $sourceIds = $connection->fetchCol($select);
```
Need to check this condition
If we do not have enabled sources then we need to return empty data without extra queries

Commit:
MSI magento-engcom/magento#65:
- Add new test case for all sources are disabled
- Simplify test case to  new structure

Commit:
Merge remote-tracking branch 'origin/develop' into msi-out-of-stock

Commit:
MSI: MSI magento-engcom/magento: #76

-- global refactoring

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento-engcom/magento2?pullRequest=76) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-engcom/magento2?pullRequest=76) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: larsroettig<br/>:x: naydav

Commit:
MSI: MSI magento-engcom/magento: #76

-- global refactoring

Commit:
MSI: MSI magento-engcom/magento: #76

-- internal build fixes
